STYLE_GUIDE.md

```diff
@@ -0,0 +1,183 @@
+- **SCSS** files use **kebab-case** (`initial-variables.scss`)
+- **JSON** files use **snake_case** (`all_lang.json`) while **Markdown** files use **SCREAMING_SNAKE_CASE** (`CONTRIBUTING.md`)
+
+## SFC Conventions
+### Skeleton
+99% of the time your SCSS should be **scoped**, so it won't bleed outside of your component and pollute the global namespace!
+```vue
+<template>
+    <div>
+        ...
+    </div>
+</template>
+
+<script lang="ts">
+import { Component, Vue } from 'nuxt-property-decorator'
+import { ComponentOne } from '@/components/ComponentOne.vue'
+
+interface InterfaceName {
+  property: type
+}
+
+@Component({
+  components: {
+    ComponentOne
+  },
+})
+export default class ComponentName extends Vue {}
+</script>
+
+<style scoped lang="scss"></style>
+```
+
+### Property Decorators
+We rely on the package 'nuxt-property-decorator', hence, we urge you to comply with the [Nuxt Class Component Syntax](https://github.com/nuxt-community/nuxt-property-decorator/)
+```typescript
+import {
+  Component,
+  Inject,
+  Model,
+  Prop,
+  Provide,
+  Vue,
+  Watch,
+} from "nuxt-property-decorator"
+
+@Component({})
+export class MyComponent extends Vue {
+  @Inject() foo!: string
+
+  @Model("change") checked!: boolean
+
+  @Prop() propA!: number
+
+  @Provide() foo = "foo"
+
+  @Watch("child")
+  onChildChanged(val: string, oldVal: string) {}
+}
+```
+
+
+### Using Components In Templates
+Custom components and prop bindings should be used like this
+```vue
+<MyCustomComponent :property-name="doSomething">
+    {{ someProperty }}
+</MyCustomComponent>
+```
+
+Use shorthands for vue attributes
+- `:href` instead of `v-bind:href`
+- `@click` instead of `v-on:click`
+- ...
+
+
+### Fetching Data
+Though we haven't yet transitioned most of our data fetching logic to Nuxt lifecycles, the following syntax should be considered best practice:
+```typescript
+// pages
+import { Component } from 'nuxt-property-decorator'
+
+@Component({
+  async asyncData({ app, $config, params }) {
+    const res = await app?.apolloProvider?.clients[$config.prefix].query({
+      query: queryGql,
+      variables: {
+        id: params.id
+      },
+    })
+
+    return {
+      data: res.data,
+      total: res.total,
+    }
+  }
+})
+export default class ClassName extends Vue {
+  data?: Type
+  total?: Type
+
+  [...]
+}
+```
+
+
+### Reusability Through Abstraction
+If your component will be used on several occasions in many different contexts, you should think about the way you pass data to your components and the way events are handled.
+In regards to event handling, you should always aim to emit events happening in your component, while the handling should be done in the parent or page itself. Thereby, the click of a button can trigger all kinds of functionality, without having to be aware of its context.
+```vue
+<template>
+    <!-- ParentComponent.vue -->
+    <ChildComponent @button-clicked="sendFormRequest" />
```

Comment:
By standard I'll try to mimic the existing event `@click @change` 
then I'll try to be self-descriptive like `@remove`

Reply:
yes i have now made the example more clear